xbmc/filesystem/HTTPFile.cpp

```diff
@@ -47,7 +47,7 @@ ssize_t CHTTPFile::Write(const void* lpBuf, size_t uiBufSize)
     return -1;
 
   std::string myPostData((char*) lpBuf);
```

Comment:
You weren't responsible for this line but I just noticed that it's casting away the const from lpBuf which it really shouldn't do :)